### ISSUE TITLE
Bug: Download via URL sticks at 0%

### ISSUE THREAD

**Issue by bensaufley** [TortugaPower/BookPlayer#925]
## Environment

 - OS Version: iOS 16.5 (20F66)
 - App Version: 5.0.0-10

## Description

First off: love Bookplayer, psyched about the major version release. Thanks for all you do!

Using `bookplayer://download` URLs to download books from a networked location appears to consistently result in a download freezing at `0%` and never resolving.

I have audiobooks from Audible, Kobo, Libro.fm, and I built a catalog that runs as a Docker container on my NAS – repo [here](https://github.com/bensaufley/audiobook-catalog) – so I can have all my audiobooks in one place and, e.g., check if I have it before I buy one on sale 😅, but also download directly to my phone easily from a graphic UI rather than messing with networking stuff.

Since the 5.0 update, all downloads just stay at `0%`. This was not true in earlier versions.

**Possible complication**: I have normally accessed the catalog on my NAS in the browser at `http://tower.local:6451` (it's Unraid), but something recently changed and now I have to access it by IP (`http://192.168.###.###:6451`). I have confirmed that the `bookplayer://` URL is showing the IP, and that IP is accessible from my phone on the WiFi. Here's an example URL:

```
bookplayer://download?url=%22http://192.168.6.205:6451/books/4eac2412-6dbc-4266-b240-5b12872f8748/download%22
```

I have used the Files app on my phone to download from the NAS to local, then from local used the Share sheet to open in Bookplayer. So this confirms that:

a. Import _does_ work in some circumstances
b. My phone can reach the NAS at the given IP

## Steps to reproduce the behavior
1. Open a `bookplayer://download?url=` URL in Bookplayer 5.x
2. Go to Bookplayer
3. Wait – the `0%` number never changes

**Comment by GianniCarlo** [comment]
Hey @bensaufley ! thanks for reporting the issue and giving a thorough description. I'll add this to be looked into in the next patch release (v5.0.2), in the meantime, could you try the new 'Download from URL' option from the import action sheet? this new option works a little different than what we have for URL scheme actions, so it'll be good to know if this one also has the same problem

![IMG_D97AD1AE0AF4-1](https://github.com/TortugaPower/BookPlayer/assets/14112819/380a29c7-d66d-4c24-a423-a36618f420ea)

**Comment by bensaufley** [comment]
@GianniCarlo ooh, nice! Does the URL _have_ to be https? Since I'm just on a local NAS I haven't bothered with it. I tried putting one of these URLs in there and it just … didn't do anything. No error, but no downloading UI either. 🤔

**Comment by GianniCarlo** [comment]
Hmmm we do have enabled to make requests to http, so that's something to look into as well then, thanks!

<img width="289" alt="Screenshot 2023-06-21 at 09 54 49" src="https://github.com/TortugaPower/BookPlayer/assets/14112819/5fff2949-9f26-47f6-bb64-15230fb04e1f">

**Comment by bensaufley** [comment]
Were you able to figure out what was going on? Interestingly enough, I tried another download today and it worked fine, several times.

But! Here's a quirk: the book I had downloaded via Files and imported that way … was gone when I looked today. I hadn't yet started it so I don't know when it vanished. I checked the "manage files" section of Settings and it wasn't there either. I guess it may be my fault: after I imported to Bookplayer, I saw that the file showed up inside the Bookplayer Cloud folder, and so I deleted from the _downloaded_ spot (not the Bookplayer directory) in Files; would that cause it to be removed from the library?

**Comment by GianniCarlo** [comment]
@bensaufley there's a new Tesflight build that have the new changes (link to join: https://testflight.apple.com/join/xLDneJXN). What I found there was that the code to handle single downloads wasn't showing any feedback when it failed, so I reworked the old network interface, and it's now using the new one we introduced for sync operations. Also another thing that could have been compounded the issue, is that there was a race condition on launch, that could prevent the import screen from showing up at any point in the app. Both changes are in the beta, so let me know if this issue happens again.

Regarding the book that vanished, hmmm it shouldn't affect that you deleted the book from outside the BookPlayer directory on Files, when you checked on the BookPlayer folder, was the book at the root level? or inside the Processed folder
